node_io/lib/src/directory.dart

```diff
@@ -39,17 +40,18 @@ class Directory extends FileSystemEntity implements io.Directory {
       FileStat.statSync(path).type == io.FileSystemEntityType.directory;
 
   @override
-  Future<io.Directory> delete({bool recursive: false}) {
+  Future<Directory> delete({bool recursive: false}) {
```

Comment:
I'd leave `io.Directory` as return type if it's not causing any issues.

Reply:
Actually it should even be io.FileSystemEntity. For delete we don't really care but for other having the proper return type allow writing without casting things such as:
```dart
var directory = new Directory(path);
directory = await directory.create();
```

If you still want to change the return type,  ok for `io.FileSystemEntity`? (However when overloading it is fine for me to override the return type with a su)

Reply:
Sounds good to me.